pkg/config/analysis/local/istiod_analyze.go

```diff
@@ -224,13 +222,29 @@ func (sa *IstiodAnalyzer) SetSuppressions(suppressions []AnalysisSuppression) {
 	sa.suppressions = suppressions
 }
 
+// AddTestReaderKubeSource adds a yaml source to the analyzer, which will analyze
+// runtime resources like pods and namespaces for use in tests.
+func (sa *IstiodAnalyzer) AddTestReaderKubeSource(readers []ReaderSource) error {
+	return sa.addReaderKubeSourceInternal(readers, true)
+}
+
 // AddReaderKubeSource adds a source based on the specified k8s yaml files to the current IstiodAnalyzer
 func (sa *IstiodAnalyzer) AddReaderKubeSource(readers []ReaderSource) error {
+	return sa.addReaderKubeSourceInternal(readers, false)
```

Comment:
Additionally, based on my understanding, this will result in the removal of some analyzers in file-only analysis that require these known types as metadata inputs. Is this the expected behavior?

Reply:
That is correct.  Analyzers which require secrets, services, pods namespaces, or nodes will be skipped when run against files, as these resources are not expected to exist in file format, and are generally created at runtime.  As far as I can tell, this was the behavior prior to last fall when I removed the galley codebase, but I made a mistake in my refactor, which caused these analyzers to be included when running against files.  This PR fixes this bug.  

Do you feel that some of these analyzers ought to run against files?  If so, which ones?

Reply:
Thank you for the explanation; this makes sense to me. I initially thought the old behavior was the same as after the removal of the galley code. I'm going to close mine, as I believe this one is much better.